qiskit_experiments/library/randomized_benchmarking/create_clifford_map.py

```diff
@@ -0,0 +1,194 @@
+as numbers. For example, if Clifford1.compose(Clifford2) == Clifford3, then we conceptually,
+we store {(1, 2) : 3}. we don't actually store the map, but only the results of the compose in an array.
+This is more efficient in performance. The result is found using the indices of the input Cliffords.
+Similarly, we store for each number representing a Clifford, the number representing the
+inverse Clifford.
+For compose, we don't actually store the full compose table of all-cliffords X all-cliffords.
+Instead, we define an array of single-gate-cliffords. This comprises all Cliffords that consist
+of a single gate. There are 8 such Cliffords for 1-qubit, and 20 such Cliffords for 2-qubits.
+It is sufficient to store the compose table of all-cliffords X single-gate-cliffords,
+since for every Clifford on the right hand side, we can break it down into single gate Cliffords,
+and do the composition one at a time. This greatly reduces the storage space for the array of
+composition results (from O(n^2) to O(n)), where n is the number of Cliffords.
+"""
+import itertools
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.library import (
+    IGate,
+    HGate,
+    SXdgGate,
+    SGate,
+    XGate,
+    SXGate,
+    YGate,
+    ZGate,
+    SdgGate,
+    CXGate,
+)
+from qiskit.quantum_info.operators.symplectic import Clifford
+from qiskit_experiments.library.randomized_benchmarking.clifford_utils import CliffordUtils
+
+gate_list_1q = [
+    IGate(),
+    HGate(),
+    SXdgGate(),
+    SGate(),
+    XGate(),
+    SXGate(),
+    YGate(),
+    ZGate(),
+    SdgGate(),
+]
+num_to_cliff_1q = {}
+cliff_to_num_1q = {}
+num_to_cliff_2q = {}
+cliff_to_num_2q = {}
+
+
+def gen_nums_single_gate_cliffs_1q():
+    """
+    We generate an array mapping numbers to Cliffords and the reverse array.
+    Based on this array, we build a mapping from every single-gate-clifford to its number.
+    The mapping actually looks like {(gate, '[0]'): num}, where [0] represents qubit 0.
+    The qubit is added to be consistent with the format for 2 qubits.
+    """
+    for i in range(CliffordUtils.NUM_CLIFFORD_1_QUBIT):
+        cliff = CliffordUtils.clifford_1_qubit(i)
+        num_to_cliff_1q[i] = cliff
+        cliff_to_num_1q[cliff.__repr__()] = i
+    clifford_single_gate_to_num = {}
+
+    for gate in gate_list_1q:
+        qc = QuantumCircuit(1)
+        qc.append(gate, [0])
+        cliff = Clifford(qc)
+        if cliff.__repr__() in cliff_to_num_1q.keys():
+            num = cliff_to_num_1q[cliff.__repr__()]
+            # qubit_as_str is not really necessary. It is only added to be consistent
+            # with the representation for 2 qubits
+            qubit_as_str = "[0]"
+            clifford_single_gate_to_num[(gate.name, qubit_as_str)] = num
+        else:
+            print("not found")
+    file.write(f"CLIFF_SINGLE_GATE_MAP_1Q = {clifford_single_gate_to_num}\n")
```

Comment:
Seems like `file` is not defined within current scope. Also you should NOT induce strong coupling to file system. This makes code unreadable, and also hardly guarantees multiple platform support. 

By the way is there any special reason to generate text data? This is expensive to load because of the deserialization overhead.

Reply:
1. Do you think it would be better if the user defined the path to the required files?
2. What do you suggest instead of text data?

Reply:
1. Some tmp file dir, or qiskit experiments data dir (e.g. in app data in Mac) -- at least somewhere not in the package. Note that qiskit provider code is kind of doing this, i.e. it saves token. However this should be usually avoided (local file system).
2. Depends on data structure.